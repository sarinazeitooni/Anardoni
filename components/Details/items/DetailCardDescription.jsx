import { useSelector , useDispatch } from "react-redux";
import style from './scss/DetailCardDescription.module.scss';
import messages from "./messages/messages";
import _ from "lodash";
const DetailCardDescription = ({item})=>{
    const description = _.get(item, "description",'');
    const more = _.get(item, "more",['']);
    const about = _.get(item, "about",['']);
    const btnValue = useSelector(state => state.btnValue);
    const dispatch = useDispatch();
    function btnToggle() {
        dispatch({
            type: "btnToggle"
        });
        document.getElementById('more-mask').style.height="fitContent";
    };
    return (
        <div>
            <div className={style.preview}>
                <span className={style['preview-title']}>{messages.description}</span>
                <div className={style['about-items-container']}>
                    {description && <div>{description}</div>}
                    {about.map((item) =>
                        <li className={style['description-items']}>
                            {item}
                        </li>
                    )}
                    <div className={style['more-container']}>
                        <div className={style['item-container']} style={{height : btnValue ? "13px" : "fit-content"}} id='more-mask'>
                            {more.map((item) => {
                                return (
                                    <li style={{color: btnValue ? "gray" : "white"}} className={style['about-items']}>
                                        {item}
                                    </li>
                                )
                            })}
                        </div>
                        <div className={style['button-container']}>
                            <button className={style['toggle-btn']} onClick={btnToggle}>{btnValue ? messages.more : messages.close }</button>
                        </div>
                    </div>
                </div>
            </div>
            <div className="line"></div>
        </div>
    )
}
export default DetailCardDescription;