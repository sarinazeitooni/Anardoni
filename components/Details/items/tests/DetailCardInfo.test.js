import React from 'react';
import { shallow } from 'enzyme';
import DetailCardDescription from "../DetailCardInfo";
describe(" DetailCardDescription component" , ()=>{
    it("it should render a div",()=>{
        const container = shallow(<DetailCardDescription/>)
        expect(container).toMatchSnapshot();
    })
});