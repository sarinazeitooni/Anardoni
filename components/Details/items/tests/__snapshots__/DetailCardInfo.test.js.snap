// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` DetailCardDescription component it should render a div 1`] = `ShallowWrapper {}`;
